### PR TITLE
pythran 0.15.0

### PR DESCRIPTION
**Pull request by timsutton**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by timsutton** [comment]
I think this one's encountering an issue related to distutils-related workarounds:

```
2024-01-14T01:56:46.9951354Z The formula built, but is not symlinked into /home/linuxbrew/.linuxbrew
2024-01-14T01:56:46.9951618Z [31mError:[0m The `brew link` step did not complete successfully
2024-01-14T01:56:46.9951936Z Could not symlink lib/python3.11/site-packages/_distutils_hack/__init__.py
2024-01-14T01:56:46.9952340Z Target /home/linuxbrew/.linuxbrew/lib/python3.11/site-packages/_distutils_hack/__init__.py
2024-01-14T01:56:46.9952481Z already exists. You may want to remove it:
2024-01-14T01:56:46.9952935Z   rm '/home/linuxbrew/.linuxbrew/lib/python3.11/site-packages/_distutils_hack/__init__.py'
```

File: Formula/p/pythran.rb [hunk 1]
```diff
@@ -38,6 +37,11 @@ class Pythran < Formula
     sha256 "9c270fe5f4b130969b54174de7db4e764b09b4f7f67ccfc32480e29f78348d97"
   end
 
+  resource "setuptools" do
+    url "https://files.pythonhosted.org/packages/fc/c9/b146ca195403e0182a374e0ea4dbc69136bad3cd55bc293df496d625d0f7/setuptools-69.0.3.tar.gz"
+    sha256 "be1af57fc409f93647f2e8e4573a142ed38724b8cdd389706a867bb4efcf1e78"
+  end
```
Comment (cho-m) [inline Formula/p/pythran.rb:43]: This is causing conflict with `depends_on "python-setuptools"`.

Due to https://github.com/Homebrew/brew/pull/16442, `setuptools` may need to be specified in 
https://github.com/Homebrew/homebrew-core/blob/c68391c81ea022b47cbe006baa8b7252873eb332/pypi_formula_mappings.json#L997-L998

Comment (chenrui333) [reply Formula/p/pythran.rb:42]: I just removed `python-setuptools` formula dependency now. relates to https://github.com/Homebrew/homebrew-core/pull/159725

**Comment by chenrui333** [comment]
```
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'setuptools'
```

**Review by chenrui333 — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]

File: Formula/p/pythran.rb (hunk 1 above)
Comment (timsutton) [reply Formula/p/pythran.rb]: Thanks, the part that had me confused was whether I still needed distutils as a depends_on, but also to exclude it in that pypi mappings file. Will confess that I don't understand much about how Brew's python distribution/packaging is wrapped at an internal level!

**Pull request merged**